src/core/events.rs

```diff
@@ -59,7 +59,7 @@ impl PartialOrd for EventKey {
 
 impl Ord for EventKey {
     fn cmp(&self, other: &Self) -> Ordering {
-        if self.time == other.time {
+        if (self.time - other.time).abs() < std::f64::EPSILON {
```

Comment:
Exact equality is expected in this if statement. It is for a comparison tiebreaker.